pyproject.toml

```diff
@@ -49,7 +49,7 @@ requests = "^2.21"
 email_validator = "^1.0"
 pydantic = "^1.4"
 typing_extensions = [
-    { version = "^3.7.4", python = "^3.6" },
+    { version = "^3.7.4", python = "~3.6" },
     { version = "^4", python = "^3.7" },
```

Comment:
It seems that you forgot the `>=` operator. 
```
{ version = "^4", python = ">=3.7" }
```

Reply:
The dependency issue is link to "^3.6" correspond to ">=3.6, <4" and "^3.7" correspond to ">=3.7, <4".  Then ">=3.6, <4" is overlapping with ">=3.7, <4"

If we replace "^" by "~" for the first one, "~3.6" correspond to ">=3.6, <3.7". 

The version are no more overlapping and the dependency issue should be gone.

Reply:
Great, thanks for the explanation !